src/courtroom.cpp

```diff
@@ -4028,6 +4031,18 @@ void Courtroom::music_synchronize(bool toggle)
     music_flags &= ~SYNC_POS;
 }
 
+void Courtroom::music_random()
+{
+  QList<QTreeWidgetItem *> clist;
+  QTreeWidgetItemIterator it(ui_music_list, QTreeWidgetItemIterator::NotHidden | QTreeWidgetItemIterator::NoChildren);
```

Comment:
Not clear how this avoids selecting categories.

Reply:
I explained it, it treats a category as an entry with any children. No children = not a category.